test/rapids_pre_commit_hooks/test_alpha_spec.py

```diff
@@ -0,0 +1,321 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from itertools import chain
+from textwrap import dedent
+from unittest.mock import Mock, call, patch
+
+import pytest
+import yaml
+
+from rapids_pre_commit_hooks import alpha_spec, lint
+
+
+@pytest.mark.parametrize(
+    ["package", "content", "mode", "replacement"],
+    [
+        *chain(
+            *(
+                [
+                    (p, p, "development", f"{p}>=0.0.0a0"),
+                    (p, p, "release", None),
+                    (p, f"{p}>=0.0.0a0", "development", None),
+                    (p, f"{p}>=0.0.0a0", "release", p),
+                ]
+                for p in alpha_spec.RAPIDS_ALPHA_SPEC_PACKAGES
+            )
+        ),
+        *chain(
+            *(
+                [
+                    (f"{p}-cu12", f"{p}-cu12", "development", f"{p}-cu12>=0.0.0a0"),
+                    (f"{p}-cu11", f"{p}-cu11", "release", None),
+                    (f"{p}-cu12", f"{p}-cu12>=0.0.0a0", "development", None),
+                    (f"{p}-cu11", f"{p}-cu11>=0.0.0a0", "release", f"{p}-cu11"),
+                ]
+                for p in alpha_spec.RAPIDS_CUDA_SUFFIXED_PACKAGES
+            )
+        ),
+        *chain(
+            *(
+                [
+                    (f"{p}-cu12", f"{p}-cu12", "development", None),
+                    (f"{p}-cu12", f"{p}-cu12>=0.0.0a0", "release", None),
+                ]
+                for p in alpha_spec.RAPIDS_ALPHA_SPEC_PACKAGES
+                - alpha_spec.RAPIDS_CUDA_SUFFIXED_PACKAGES
+            )
+        ),
+        ("cuml", "cuml>=24.04,<=24.06", "development", "cuml<=24.06,>=0.0.0a0,>=24.04"),
+        ("cuml", "cuml>=24.04,<=24.06,>=0.0.0a0", "release", "cuml<=24.06,>=24.04"),
+        (
+            "cuml",
+            "&cuml cuml>=24.04,<=24.06,>=0.0.0a0",
```

Comment:
Is this testing the right thing? We shouldn't have a specifier that is inclusive of both 24.04 and 24.06.
```suggestion
        ("cuml", "cuml>=24.04,<24.06", "development", "cuml<24.06,>=0.0.0a0,>=24.04"),
        ("cuml", "cuml>=24.04,<24.06,>=0.0.0a0", "release", "cuml<24.06,>=24.04"),
        (
            "cuml",
            "&cuml cuml>=24.04,<24.06,>=0.0.0a0",
```